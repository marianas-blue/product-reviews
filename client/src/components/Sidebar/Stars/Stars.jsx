/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable react/self-closing-comp */
/* eslint-disable no-useless-constructor */
/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import { Links, Descriptor, Number, Bars, Yellow, Rating, Starbox, Tooltiptext, Tooltip } from './Stars.style';

class Stars extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Descriptor>
        <Number>453 customer reviews</Number>
        <div />
        <Starbox>
          <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png" alt="stars" height="25px" />
          {'   '}
          <Tooltip>
            4.2 out of 5 stars
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/downarrow.png" alt="down" height="10px" />
            <Tooltiptext>
            Amazon calculates a productâ€™s star ratings based on a
            machine learned model instead of a raw data average. The
            model takes into account factors including the age of a rating,
            whether the ratings are from verified purchasers, and factors that
            establish reviewer trustworthiness.
            </Tooltiptext>
          </Tooltip>
        </Starbox>

        <div>
          <Links href="#top">
          5 star
            {'    '}
            <Bars>
              <Yellow></Yellow>
            </Bars>
          </Links>
          <Rating>20%</Rating>
        </div>
        <div>
          <Links href="#top">
          4 star
            {'    '}
            <Bars>
              <Yellow></Yellow>
            </Bars>          
          </Links>
          <Rating>20%</Rating>
        </div>
        <div>
          <Links href="#top">
          3 star
            {'    '}
            <Bars>
              <Yellow></Yellow>  
            </Bars>          
          </Links>
          <Rating>20%</Rating>
        </div>
        <div>
          <Links href="#top">
          2 star
            {'    '}
            <Bars>
              <Yellow></Yellow>  
            </Bars>          
          </Links>
          <Rating>20%</Rating>
        </div>
        <div>
          <Links href="#top">
          1 star
            {'    '}
            <Bars>
              <Yellow></Yellow>
            </Bars>
          </Links>
          <Rating>20%</Rating>
        </div>
      </Descriptor>
    );
  }
}

export default Stars;
