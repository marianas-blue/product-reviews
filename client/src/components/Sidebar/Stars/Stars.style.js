import styled from 'styled-components';

export const Links = styled.a`
color: #0066c0;
display: inline block;
padding: 5px 10px 5px 10px;
text-decoration: none;
:hover {
  text-decoration: underline;
  color: #c60;
}
`;

export const Descriptor = styled.div`
padding: 2px;
font-size: 14px;
padding: 0px 15px 0px 0px;
`;

export const Number = styled.button`
color: #c76800;
font-size: 17px;
font-weight: bolder;
border: none; 
padding: 0;
background-color: transparent;
outline: none;
text-align: left;
:hover {
  color: #c60;
}
`;

export const Bars = styled.div`
background-color: #f1f1f1;
width: 195px;
height: 17px;
display: inline-block;
border: 1px solid #b1b1b1;
`;

export const Yellow = styled.div`
background-color: #ffc200;
width: 39px;
height:17px;
border-radius: 2px;
background: linear-gradient(to bottom,#ffce00,#ffa700);
background-color: #ffce00;
border-right-color: #cc9204;
`;

export const Rating = styled.div`
color: #797979;
display: inline-block;
`;

export const Starbox = styled.div`
display: flex;
align-items: center;
margin-top: 7px;
margin-bottom: 15px;
justify-content: flex-start;
`;

export const Tooltiptext = styled.div`
  visibility: hidden;
  width: 400px;
  background-color: white;
  color: black;
  text-align: center;
  border-radius: 3px;
  padding: 10px;
  border: 1px solid #f1f3f5;
  position: absolute;
  z-index: 1;
`;

export const Tooltip = styled.div`
position: relative;
display: inline-block;

&:hover ${Tooltiptext} {
  visibility: visible;
}
`;