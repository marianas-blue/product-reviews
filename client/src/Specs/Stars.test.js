/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import Stars from '../components/Sidebar/Stars/Stars';

describe('Stars', () => {
  it('should render correctly', () => {
    const component = shallow(<Stars />);

    expect(component).toMatchSnapshot();
  });
});
