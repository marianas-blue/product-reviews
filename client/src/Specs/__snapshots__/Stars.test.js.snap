// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars should render correctly 1`] = `
<styled.div>
  <styled.button>
    453 customer reviews
  </styled.button>
  <div />
  <styled.div>
    <img
      alt="stars"
      height="25px"
      src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png"
    />
       
    <styled.div>
      4.2 out of 5 stars
      <img
        alt="down"
        height="10px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/downarrow.png"
      />
      <styled.div>
        Amazon calculates a productâ€™s star ratings based on a machine learned model instead of a raw data average. The model takes into account factors including the age of a rating, whether the ratings are from verified purchasers, and factors that establish reviewer trustworthiness.
      </styled.div>
    </styled.div>
  </styled.div>
  <div>
    <styled.a
      href="#top"
    >
      5 star
          
      <styled.div>
        <styled.div />
      </styled.div>
    </styled.a>
    <styled.div>
      20%
    </styled.div>
  </div>
  <div>
    <styled.a
      href="#top"
    >
      4 star
          
      <styled.div>
        <styled.div />
      </styled.div>
    </styled.a>
    <styled.div>
      20%
    </styled.div>
  </div>
  <div>
    <styled.a
      href="#top"
    >
      3 star
          
      <styled.div>
        <styled.div />
      </styled.div>
    </styled.a>
    <styled.div>
      20%
    </styled.div>
  </div>
  <div>
    <styled.a
      href="#top"
    >
      2 star
          
      <styled.div>
        <styled.div />
      </styled.div>
    </styled.a>
    <styled.div>
      20%
    </styled.div>
  </div>
  <div>
    <styled.a
      href="#top"
    >
      1 star
          
      <styled.div>
        <styled.div />
      </styled.div>
    </styled.a>
    <styled.div>
      20%
    </styled.div>
  </div>
</styled.div>
`;
